model/ddl.go

```diff
@@ -74,19 +75,37 @@ type Job struct {
 	TableID  int64         `json:"table_id"`
 	State    JobState      `json:"state"`
 	Error    *terror.Error `json:"err"`
-	// every time we meet an error when running job, we will increase it
-	ErrorCount int64         `json:"err_count"`
-	Args       []interface{} `json:"-"`
+	// every time we meet an error when running job, we will increase it.
+	ErrorCount int64 `json:"err_count"`
+	// the number of rows are processed.
+	RowCount int64         `json:"row_count"`
+	Mu       sync.Mutex    `json:"-"`
```

Comment:
Because it's the serial call `GetRowCount` and `SetRowCount`.  Call `Encode` and `SetRowCount` are asynchronous, but only put the job to queue will call `Encode`(per max(2*lease, 10s)).